fog/ledger/server/src/router_handlers.rs

```diff
@@ -332,7 +332,7 @@ async fn authenticate_ledger_store<E: LedgerEnclaveProxy>(
     ledger_store_url: KeyImageStoreUri,
     logger: Logger,
 ) -> Result<(), RouterServerError> {
-    let ledger_store_id = ResponderId::from_str(&ledger_store_url.to_string())?;
+    let ledger_store_id = ledger_store_url.responder_id()?;
```

Comment:
I'd like to discuss this a bit more. The difference in terms of result between these two lines is that the old one includes the protocol specifier, `insecure-key-image-store://`, and the second one doesn't. To me, it seems like either of these should work. However, what we actually got is that the Noise AEAD decryption failed on the response from the Key Image Store to the router.

Reply:
The `fn responder_id` method grabs the query param from the URL and passes it to the lower-level `from_str` method. There a few methods that grab query params and are named similarly (e.g. `fn consensus_msg_key`). It is a bit confusing, agreed, but I think the solution might involve renaming methods to denote that we are getting the values from a query param. So instead of `fn responder_id` it would be `fn responder_id_from_param` or something less verbose but still indicating that we got it from a query param.

What do you think?

Reply:
Honestly, the issue here is "what is a responder ID" in general, IMO. The name is not very explanatory and since it's only constructed from a string there's not a lot of validation on whether a given ID is acceptable or not.

But James assures me that responder IDs will go away when we move to DCAP, so I'm tempted to just not worry about it.